--- doomlegacy-1.44/src/linux_x/i_video_xshm.c~	2011-01-20 03:01:17.000000000 +0200
+++ doomlegacy-1.44/src/linux_x/i_video_xshm.c	2011-01-20 03:02:46.595584187 +0200
@@ -1626,14 +1626,14 @@
     if(rendermode==render_soft) {
     // setup attributes for main window
     if (vidmode_active) {
-       attribmask = CWColormap | CWSaveUnder | CWBackingStore |
+       attribmask = CWSaveUnder | CWBackingStore |
           CWEventMask | CWOverrideRedirect;
 
        attribs.override_redirect = True;
        attribs.backing_store = NotUseful;
        attribs.save_under = False;
     } else
-       attribmask = CWBorderPixel | CWColormap | CWEventMask;
+       attribmask = CWBorderPixel | CWEventMask;
 
     attribs.event_mask = KeyPressMask | KeyReleaseMask
 #ifndef POLL_POINTER
@@ -1641,7 +1641,10 @@
 #endif
        | ExposureMask | StructureNotifyMask;
 
-    attribs.colormap = X_cmap;
+    if (x_pseudo) {
+        attribmask |= CWColormap;
+        attribs.colormap = X_cmap;
+    }
     attribs.border_pixel = 0;
 
     // create the main window
