--- doomlegacy-1.32/doomlegacy/linux_x/musserv/sequencer.c.orig	Fri May 11 14:55:27 2001
+++ doomlegacy-1.32/doomlegacy/linux_x/musserv/sequencer.c	Wed Jul 18 17:26:02 2001
@@ -176,11 +176,7 @@
   seq_dev = num_fm;
 
   sndstat = fopen("/dev/sndstat", "r");
-  if (sndstat == NULL)
-    {
-    printf("musserver: could not open /dev/sndstat, exiting.\n");
-    exit(1);
-    }
+  if (sndstat != NULL) {
   fgets(sndver, 100, sndstat);
   for (x = 0; x < strlen(sndver); x++)
     if (sndver[x] == '-')
@@ -198,7 +194,9 @@
       }
   voxdate = atoi(snddate);
   fclose(sndstat);
-
+  } else {
+    voxdate = 0; /* kernel 2.4 doesn't have sndstat */
+  }
   voices = malloc(sinfo[seq_dev].nr_voices * sizeof(struct synth_voice));
   for (x = 0; x < sinfo[seq_dev].nr_voices; x++)
     {
