--- doomlegacy-1.44/src/linux_x/musserv/Makefile.linux~	2009-09-24 08:04:44.000000000 +0300
+++ doomlegacy-1.44/src/linux_x/musserv/Makefile.linux	2011-01-20 02:50:42.858632376 +0200
@@ -3,7 +3,7 @@
 #
 
 CC	= gcc
-CFLAGS	= -I. -Wall -O2 -march=pentium
+CFLAGS	= -I. -Wall $(OPTFLAGS)
 LDFLAGS	=
 #LDFLAGS	= -static
 
--- doomlegacy-1.44/src/linux_x/sndserv/Makefile~	2009-09-24 08:04:44.000000000 +0300
+++ doomlegacy-1.44/src/linux_x/sndserv/Makefile	2011-01-20 02:52:40.272020598 +0200
@@ -30,7 +30,7 @@
 #
 
 CC=gcc
-CFLAGS=-O2 -march=pentium -Wall -DNORMALUNIX -DLINUX
+CFLAGS=$(OPTFLAGS) -Wall -DNORMALUNIX -DLINUX
 ifdef FREEBSD
 CFLAGS:=${CFLAGS} -DFREEBSD
 endif
--- doomlegacy-1.42/doomlegacy_142_src/tools/makefile.orig	Sat Feb 22 21:13:11 2003
+++ doomlegacy-1.42/doomlegacy_142_src/tools/makefile	Sat Feb 22 21:14:30 2003
@@ -1,24 +1,24 @@
 
 default:    dircomp2.c
-	gcc -O6 -mpentium -Wall -s dircomp2.c -o dircomp2.exe
+	$(CC) $(OPTFLAGS) -Wall dircomp2.c -o dircomp2.exe
 
 asm:
-	gcc -O6 -mpentium -Wall -g dircomp2.c -o tmp.exe
+	$(CC) $(OPTFLAGS) -Wall -g dircomp2.c -o tmp.exe
 	objdump  -dS tmp.exe --debugging --no-show-raw-insn > dircomp2.s
 	del tmp.exe
 
 h2d:	 h2d.c
-	gcc -O6 -mpentium -Wall -s h2d.c -o h2d.exe
+	$(CC) $(OPTFLAGS) -Wall h2d.c -o h2d.exe
 
 dckconv:     dckconv.c
-	gcc -O6 -mpentium -Wall -s dckconv.c -o dckconv.exe
+	$(CC) $(OPTFLAGS) -Wall dckconv.c -o dckconv.exe
 
 dckconv2:     dckconv2.c
-	gcc -O6 -mpentium -Wall -s dckconv2.c -o dckconv2.exe
+	$(CC) $(OPTFLAGS) -Wall dckconv2.c -o dckconv2.exe
 
 dckcomp:     dckcomp.c
-	gcc -O6 -mpentium -Wall -s dckcomp.c -o dckcomp.exe
+	$(CC) $(OPTFLAGS) -Wall dckcomp.c -o dckcomp.exe
 
 convert:    convert.c
-	gcc -O6 -mpentium -Wall -s convert.c -o convert.exe
+	$(CC) $(OPTFLAGS) -Wall convert.c -o convert.exe
 	
--- doomlegacy-1.44/src/Makefile	2011-01-20 12:36:18.696704389 +0200
+++ doomlegacy-1.44/src/Makefile	2011-01-20 12:39:25.345251712 +0200
@@ -57,6 +57,9 @@
 OPTLEV = -O3
 #OPTLEV = -O5
 
+# OPTFLAGS together, not just opt level
+OPTFLAGS = $(OPTLEV)
+
 # Debug enables, save on reinventing this everytime.
 #DEBUG=1
 #PROFILEMODE=1
@@ -78,6 +81,7 @@
 
 # gcc or g++
 CC=gcc
+CXX=g++
 
 # End of User tunable settings
 
@@ -147,8 +151,7 @@
   X=1
 
   ifdef X
-    OPTS := $(OPTS) $(POLL_POINTER) -I. -I/usr/X11R6/include
-    LDFLAGS=-L/usr/X11R6/lib
+    OPTS := $(OPTS) $(POLL_POINTER) -I.
     LIBS=-lXext -lX11 -lm -ldl -lXxf86vm
 
     #WITH_DGA=1
@@ -202,8 +205,7 @@
   endif
 
   ifdef X
-    OPTS := $(OPTS) $(POLL_POINTER) -I. -I/usr/X11R6/include
-    LDFLAGS=-L/usr/X11R6/lib
+    OPTS := $(OPTS) $(POLL_POINTER) -I.
     LIBS=-lXext -lX11 -lm -lXxf86vm -lipx -lkvm -pthread
     # name of the exefile
     EXENAME=llxdoom
@@ -213,7 +215,7 @@
       $(O)/hw_md2.o $(O)/hw_cache.o $(O)/hw_trick.o
 
   else #ifdef X
-    OPTS := $(OPTS) -DVID_GGI -I. -I/usr/local/include
+    OPTS := $(OPTS) -DVID_GGI -I.
     LDFLAGS=
     LIBS=-lggi -lm -lipx
     # name of the exefile
@@ -294,7 +296,7 @@
 
 ifdef PROFILEMODE
 # build with gprof profiling information
-  CFLAGS += -g -pg -fomit-frame-pointer
+  CFLAGS += -g -pg
 else
 ifdef DEBUG
 # No -fomit-frame-pointer because debugging on some machines depends upon it.
@@ -303,7 +305,7 @@
 else
 # build a normal optimized version
   #CFLAGS += -O3
-  CFLAGS += $(OPTLEV) -fomit-frame-pointer
+  CFLAGS += $(OPTFLAGS)
 endif
 endif
 
@@ -464,13 +466,13 @@
 
 
 $(BIN)/wadtool: $(O)/wadtool.o $(O)/md5_cpp.o
-	g++ $(LDFLAGS) $^ -o $@
+	$(CXX) $(LDFLAGS) $^ -o $@
 
 $(O)/wadtool.o: common/tools/wadtool.cpp
-	g++ $(CFLAGS) -std=c++0x -c $< -o $@
+	$(CXX) $(CFLAGS) -std=c++0x -c $< -o $@
 
 $(O)/md5_cpp.o: md5.c
-	g++ $(CFLAGS) -std=c++0x -c $< -o $@
+	$(CXX) $(CFLAGS) -std=c++0x -c $< -o $@
 
 wad	: $(BIN)/wadtool
 ifdef WAD
@@ -507,7 +509,7 @@
 
 #dll
 dll : $(O)/r_opengl.o $(O)/ogl_x11.o
-	$(CC) -O3 -o $(BIN)/r_opengl.so -shared -nostartfiles $(O)/r_opengl.o $(O)/ogl_x11.o -L/usr/X11R6/lib -L/usr/local/lib -lX11 -lXext -lGL -lGLU -lm
+	$(CC) $(OPTFLAGS) -o $(BIN)/r_opengl.so -shared -nostartfiles $(O)/r_opengl.o $(O)/ogl_x11.o -lX11 -lXext -lGL -lGLU -lm
 
 musserv:
 	@mkdir -p $(MUSSERV)/linux
@@ -533,7 +535,7 @@
 
 
 $(O)/ogl_x11.o:  hardware/r_opengl/ogl_x11.c hardware/r_opengl/r_opengl.h
-	$(CC) $(CFLAGS) -shared -nostartfiles -I/usr/X11R6/include -c $< -o $@ 
+	$(CC) $(CFLAGS) -shared -nostartfiles -c $< -o $@ 
 
 
 #=======================================================
