i_sound.c needs -lm for pow()

--- doomlegacy-1.44/src/Makefile~	2011-03-03 13:06:33.000000000 +0200
+++ doomlegacy-1.44/src/Makefile	2011-03-03 13:07:20.011250325 +0200
@@ -122,7 +122,7 @@
   else
     # default is Linux, for all unix SDL
     OPTS := -DLINUX
-    LIBS := -lGL -lGLU -L/usr/X11R6/lib
+    LIBS := -lGL -lGLU -L/usr/X11R6/lib -lm
     # -L/usr/X11R6/lib is needed by Linux 2.4 and others that still have
     # the GLU libraries in an X11 directory.
   endif
